tests/expectations/test_expectation_atomic_renderers.py

```diff
@@ -97,12 +85,1077 @@ def _get_diagnostic_rendered_content(
+        "name": "atomic.prescriptive.summary",
+        "value": {
+            "params": {
+                "column_set": {"schema": {"type": "array"}, "value": ["a", "b", "c"]},
+                "column_set_0": {"schema": {"type": "string"}, "value": "a"},
+                "column_set_1": {"schema": {"type": "string"}, "value": "b"},
+                "column_set_2": {"schema": {"type": "string"}, "value": "c"},
+                "exact_match": {"schema": {"type": "boolean"}, "value": True},
+            },
+            "schema": {"type": "com.superconductive.rendered.string"},
+            "template": "Must have exactly these columns (in any order): $column_set_0 $column_set_1 $column_set_2",  # noqa: E501
+        },
+        "value_type": "StringValueType",
+    }
+
+    snapshots["test_atomic_prescriptive_summary_expect_table_row_count_to_be_between"] = {
+        "name": "atomic.prescriptive.summary",
+        "value": {
+            "params": {"min_value": {"schema": {"type": "number"}, "value": 1}},
+            "schema": {"type": "com.superconductive.rendered.string"},
+            "template": "Must have greater than or equal to $min_value rows.",
+        },
+        "value_type": "StringValueType",
+    }
+
+    snapshots["test_atomic_prescriptive_summary_expect_table_row_count_to_equal"] = {
+        "name": "atomic.prescriptive.summary",
+        "value": {
+            "params": {"value": {"schema": {"type": "number"}, "value": 10}},
+            "schema": {"type": "com.superconductive.rendered.string"},
+            "template": "Must have exactly $value rows.",
+        },
+        "value_type": "StringValueType",
+    }
+
+    snapshots["test_atomic_prescriptive_summary_expect_table_row_count_to_equal_other_table"] = {
+        "name": "atomic.prescriptive.summary",
+        "value": {
+            "params": {
+                "other_table_name": {
+                    "schema": {"type": "string"},
+                    "value": {"schema": {"type": "string"}, "value": "other_table_name"},
+                }
+            },
+            "schema": {"type": "com.superconductive.rendered.string"},
+            "template": "Row count must equal the row count of table $other_table_name.",
+        },
+        "value_type": "StringValueType",
+    }
+
+    return snapshots
+
+
+@pytest.fixture
+def assert_snapshot(atomic_renderer_snapshots: dict, request) -> Callable[[dict], None]:
+    test_name = request.node.name
+
+    def _assert_snapshot(data: dict):
+        if test_name not in atomic_renderer_snapshots:
+            pytest.fail(f"No snapshot available for test: {test_name}")
+
+        snapshot = atomic_renderer_snapshots[test_name]
+        assert snapshot == data
+
+    return _assert_snapshot
+
+
 # "atomic.prescriptive.summary" tests
 
 
 @pytest.mark.unit
 def test_atomic_prescriptive_summary_expect_column_bootstrapped_ks_test_p_value_to_be_greater_than(
-    snapshot,
+    assert_snapshot,
```

Comment:
The best approach here is to parameterize all these tests but this feels like a good step forward

Reply:
Something like this:

```python
@pytest.mark.parametrize(
    "update_dict, expected_content",
    [
        pytest.param({...}, {...}, id="expect_column_distinct_values_to_be_in_set"),
    ]
)
@pytest.mark.unit
def test_atomic_prescriptive_rendered_content(update_dict: dict, expected_content: dict, get_prescriptive_rendered_content):
    rendered_content = get_prescriptive_rendered_content(update_dict)

    actual = rendered_content.to_json_dict()
    assert actual == expected_content
```